qp_klp/tests/test_step.py

```diff
@@ -932,3 +958,237 @@ def test_precheck(self):
+        }
+
+        self.assertDictEqual(obs, exp)
+
+        # verify each prep_info_file contains the expected number of rows and
+        # more importantly are not empty.
+        for qiita_id in obs:
+            for prep_info_file in obs[qiita_id]:
+                df = parse_prep(prep_info_file)
+                self.assertEqual(10, df.shape[0])
+
+        # Now, let's fake enough of the later steps to test the loading of
+        # each prep-info file into Qiita and ensure that each post is unique
+        # from a unique file.
+
+        fake_client = FakeClient()
+        step.update_prep_templates(fake_client)
+        step.generate_special_map(fake_client)
+
+        # Since load_preps_into_qiita() relies on the hierarchy of files in
+        # QCJob()'s output to do it's work, copy the faked fastq files made
+        # for GenPrepFileJob() into the right location in QCJob's output.
+        # (We are skipping a number of steps, hence the need to do this.
+        #  Ordinarily, the data would be created in QCJob() and migrated
+        #  through the pipeline to GenPrepFileJob(), not the other way around.)
+        copytree(self.fastq_dir_13059, self.qc_fastq_dir_13059)
+        copytree(self.fastq_dir_11661, self.qc_fastq_dir_11661)
+        results = step.load_preps_into_qiita(fake_client)
+
+        occurances = {
+            'RMA_KHP_rpoS_Mage_Q97L_A7_S270_L001_R1_001.trimmed.fastq.gz': 0,
+            'RMA_KHP_rpoS_Mage_Q97E_A12_S270_L001_R2_001.trimmed.fastq.gz': 0,
+            'JBI_KHP_HGL_022_B16_S270_L001_R2_001.trimmed.fastq.gz': 0,
+            'AP581451B02_A21_S270_L001_R2_001.trimmed.fastq.gz': 0,
+            'EP159692B04_C8_S270_L001_R1_001.trimmed.fastq.gz': 0,
+            'LP127890A01_D6_S270_L001_R1_001.trimmed.fastq.gz': 0}
+
+        # take advantage of the posted-files listing being in string format
+        # and test for the appearance of specific individual file names in
+        # each post. Each fastq file should only appear in one and only one
+        # post.
+        for prep_id in fake_client.saved_posts:
+            posted_files_listing = fake_client.saved_posts[prep_id]['files']
+            for fastq in occurances:
+                if fastq in posted_files_listing:
+                    occurances[fastq] += 1
+
+        for fastq in occurances:
+            self.assertEqual(occurances[fastq], 1)
+
+        # confirm that six preps were created by confirming that
+        # load_preps_into_qiita() returned six unique prep-ids, and six
+        # unique job-ids.
+        self.assertEqual(len(results['Linking JobID'].unique()), 6)
+        self.assertEqual(len(results['Qiita Prep ID'].unique()), 6)
+
+        # confirm that three of the preps belong to the Feist project, while
+        # the other three belong to NYU.
+        projects = ['Feist_11661', 'NYU_BMS_Melanoma_13059']
+        for project in projects:
+            self.assertEqual(results['Project'].value_counts()[project], 3)
```

Comment:
Thank you for adding this. Do you think is worth checking that the `fastq.gz` files submitted to each prep are correct or is that a sure thing?

Reply:
In this test they're just text files with 'This is a text file.' inside that have been gzipped. The only things correct about them are the name and they've been gzipped; the size and content are not valuable information.

I do pick a file unique to each post and test to make sure that one and only one post contains that file; I believe that's enough of a check, given that the files are faked.